### PR TITLE
Add support for editing custom attachments

### PR DESCRIPTION
**Pull request by nuno-vieira**
### 🔗 Issue Links
Resolves https://github.com/GetStream/ios-issues-tracking/issues/463

### 🎯 Goal
Adds support for editing custom attachments. 

### 📝 Summary
A new API was added as part of the custom attachments flow. When creating the `ChatClient` it is now required to register the custom attachment payload through `ChatClient.registerAttachment(payloadType:)`.

Internally, we are currently using a static property because we can't afford having breaking changes. For StreamChat v5, we will most likely improve the attachments experience based on this WIP PR: https://github.com/GetStream/stream-chat-swift/pull/2879

### 🎨 Showcase
N/A. We will eventually add a custom attachment in the Demo App out-of-the-box once we refresh the docs around custom attachments.

### 🧪 Manual Testing Notes
N/A

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [ ] This change should be manually QAed
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [x] Affected documentation updated (docusaurus, tutorial, CMS)

### CODE REVIEW

File: Sources/StreamChat/ChatClient.swift [hunk 1]
```diff
@@ -51,6 +51,10 @@ public class ChatClient {
     /// The notification center used to send and receive notifications about incoming events.
     private(set) var eventNotificationCenter: EventNotificationCenter
 
+    /// The register that contains all the attachment payloads associated with their attachment types.
```
Comment (martinmitrevski) [inline Sources/StreamChat/ChatClient.swift:54]: you mean "registry" here?

**Review by martinmitrevski — COMMENTED** [review]
looks good! Some tests and docs would be needed before merging.

File: Sources/StreamChat/ChatClient.swift (hunk 1 above)
Comment (nuno-vieira) [reply Sources/StreamChat/ChatClient.swift:54]: Yes! Nice catch

**Review by martinmitrevski — APPROVED** [review]
LGTM! ✅

**Pull request merged**